base-notebook/Dockerfile

```diff
@@ -82,25 +83,22 @@ RUN echo "auth requisite pam_deny.so" >> /etc/pam.d/su && \
     fix-permissions $CONDA_DIR
 
 USER $NB_UID
-WORKDIR $HOME
 ARG PYTHON_VERSION=default
 
 # Setup work directory for backward-compatibility
-RUN mkdir /home/$NB_USER/work && \
-    fix-permissions /home/$NB_USER
-
-# Install conda as jovyan and check the md5 sum provided on the download site
-ENV MINICONDA_VERSION="${miniconda_version}" \
-    CONDA_VERSION="${conda_version}"
+RUN mkdir "/home/$NB_USER/work" && \
+    fix-permissions "/home/$NB_USER"
 
+# Install conda as jovyan and check the sha256 sum provided on the download site
 WORKDIR /tmp
-RUN wget --quiet https://repo.continuum.io/miniconda/Miniconda3-py38_${MINICONDA_VERSION}-Linux-x86_64.sh && \
-    echo "${miniconda_checksum} *Miniconda3-py38_${MINICONDA_VERSION}-Linux-x86_64.sh" | md5sum -c - && \
-    /bin/bash Miniconda3-py38_${MINICONDA_VERSION}-Linux-x86_64.sh -f -b -p $CONDA_DIR && \
-    rm Miniconda3-py38_${MINICONDA_VERSION}-Linux-x86_64.sh && \
+
+RUN wget --quiet "https://github.com/conda-forge/miniforge/releases/download/${miniforge_version}/${miniforge_installer}" && \
+    wget --quiet "https://github.com/conda-forge/miniforge/releases/download/${miniforge_version}/${miniforge_installer}.sha256" && \
+    sha256sum -c "${miniforge_installer}.sha256" && \
```

Comment:
I don't think it's safe to compare the sha256 of the installer with the *downloaded* sha256.

Maybe I'm getting it wrong, but if someone will have the access to the repo, they will change sha256 and we will get no error.

Reply:
@mathbunnyru I asked myself the question. I didn't find any clear recommandation. I will change it you must be right.
Thanks for the review 😄